### ISSUE TITLE
Getting `TypeError: Error processing argument -1.` when trying to play a stream with wcjs-player.

### ISSUE THREAD

**Issue by ardoramor** [jaruba/wcjs-player#31]
Here is the console.log stack:

```
Uncaught Error: Error processing argument -1.
TypeError: Error processing argument -1.
	at TypeError (native)
	at callFunction (C:\Users\...\Documents\TeraLogics\hydra\node_modules\electron-prebuilt\dist\resources\atom.asar\browser\lib\rpc-server.js:174:18)
	at EventEmitter.<anonymous> (C:\Users\...\Documents\TeraLogics\hydra\node_modules\electron-prebuilt\dist\resources\atom.asar\browser\lib\rpc-server.js:260:14)
	at emitMany (events.js:108:13)
	at EventEmitter.emit (events.js:182:7)
	at EventEmitter.<anonymous> (C:\Users\...\Documents\TeraLogics\hydra\node_modules\electron-prebuilt\dist\resources\atom.asar\browser\api\lib\web-contents.js:99:23)
	at emitTwo (events.js:87:13)
	at EventEmitter.emit (events.js:172:7)metaToValue @ C:\Users\...\Documents\TeraLogics\hydra\node_modules\electron-prebuilt\dist\resources\atom.asar\…:110

RemoteMemberFunction @ C:\Users\...\Documents\TeraLogics\hydra\node_modules\electron-prebuilt\dist\resources\atom.asar\…:144

allowSleep @ C:\Users\...\Documents\TeraLogics\hydra\build\node_modules\wcjs-player\index.js:1774

(anonymous function) @ C:\Users\...\Documents\TeraLogics\hydra\build\node_modules\wcjs-player\index.js:571
```

The stream I'm trying to play is RTMP live stream or HLS dvr stream, both give the same error.

How can this be debugged?

**Comment by jaruba** [comment]
What electron version are you using, it seems that the error is from the `power save blocker` api in electron.

https://github.com/jaruba/wcjs-player/blob/master/index.js#L1771

The way this is used in `wcjs-player`:
```
var powerSaveBlocker = require('remote').require('power-save-blocker');
var sleepId;

// prevent monitor sleep
if (!powerSaveBlocker.isStarted(sleepId))
    sleepId = powerSaveBlocker.start('prevent-display-sleep');

// allow monitor sleep
if (powerSaveBlocker.isStarted(sleepId))
    powerSaveBlocker.stop(sleepId);
```

Can you try using this code in the console log? Maybe Electron changed their API for it..

**Comment by ardoramor** [comment]
Yeah, I get the exception on the following like of the code snippet:

   if (!powerSaveBlocker.isStarted(sleepId))

Looks like the ID is expected to the an ID of a started power save blocker: https://github.com/atom/electron/blob/master/docs/api/power-save-blocker.md

I'm using version 0.34.2. I got it from vue-electron boilerplate (https://github.com/bradstewart/electron-boilerplate-vue). I can try updating the version but from the API it seems like `.isStarted` is not expecting `undefined`.

**Comment by jaruba** [comment]
You mean it works if you set:

```
var sleepId = 0;
```

as the default value?

**Comment by ardoramor** [comment]
Hey, that actually worked!

![image](https://cloud.githubusercontent.com/assets/6255652/12376180/13b56578-bcb5-11e5-9d26-efb084840a48.png)

**Comment by ardoramor** [comment]
But I'm not sure if `0` is the default. From their doc, it looks like a blocker needs to be started first, which will return an `id`. This `id` can later be used to check on the state of the blocker. I understand that the above are just snippets but maybe `sleepId` loses its value somewhere in the code and passes `undefined` to the initial call?

**Comment by jaruba** [comment]
The snippet is based on the exact way that it's used in `wcjs-player`, so the variable is created the same way:
https://github.com/jaruba/wcjs-player/blob/master/index.js#L33

I think it's fine to replace that line with:
```
    sleepId = 0;
```

In the initial check, `powerSaveBlocker.isStarted(sleepId)` should return `false` for `0` anyway, so it should be ok.

Do you want to make a PR? Or should I change it?

**Comment by ardoramor** [comment]
I'll make a PR

**Comment by ardoramor** [comment]
What's the process of releasing this as part of a new minor version of npm package?

**Comment by jaruba** [comment]
I published it on npm as `wcjs-player` v0.5.3
https://www.npmjs.com/package/wcjs-player

**Comment by ardoramor** [comment]
Thank you! It works and the issue can be closed now.
